automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/billing/BillingSpec.scala

```diff
@@ -23,110 +23,115 @@ class BillingSpec extends FreeSpec with WebBrowserSpec with UserFixtures with Cl
-                  val methodConfigDetailsPage = new WorkspaceMethodConfigDetailsPage(billingProjectName, workspaceName, SimpleMethodConfig.configNamespace, methodConfigName).open
-                  val submissionDetailsPage = methodConfigDetailsPage.launchAnalysis(MethodData.SimpleMethod.rootEntityType, TestData.SingleParticipant.entityId)
+              // verify running a method
+              val methodConfigDetailsPage = new WorkspaceMethodConfigDetailsPage(billingProjectName, workspaceName, SimpleMethodConfig.configNamespace, methodConfigName).open
+              val submissionDetailsPage = methodConfigDetailsPage.launchAnalysis(MethodData.SimpleMethod.rootEntityType, TestData.SingleParticipant.entityId)
+
+              submissionDetailsPage.waitUntilSubmissionCompletes()
+              assert(submissionDetailsPage.verifyWorkflowSucceeded())
+              //END: Test running anlysis in workspace
+
+              //BEGIN: Test adding user to project
+              billingPage.open.openBillingProject(billingProjectName)
+              billingPage.addUserToBillingProject(secondUser, "User")
+
+              val isSuccess = billingPage.isUserInBillingProject(secondUser)
+              isSuccess shouldEqual true
+              //END: Test adding user to project
 
-                  submissionDetailsPage.waitUntilSubmissionCompletes()
-                  assert(submissionDetailsPage.verifyWorkflowSucceeded())
-                }
-              }
+              //BEGIN: should be able to change the billing account associated with the project
+              val originalBillingAccount = Google.billing.getBillingProjectAccount(billingProjectName)
+              originalBillingAccount shouldBe Some(Config.Projects.billingAccountId)
+
+              Google.billing.removeBillingProjectAccount(billingProjectName)
+
+              val newBillingAccount = Google.billing.getBillingProjectAccount(billingProjectName)
+              newBillingAccount shouldBe None
+              //END: should be able to change the billing account associated with the project
             }
           }
         }
       }
     }
   }
+
+  private def createNewBillingProject(user: Credentials, billingPage: BillingManagementPage, trials: Int = 3): String = {
+    val billingProjectName = "billing-spec-create-" + makeRandomId()
+    logger.info(s"Creating billing project: $billingProjectName")
+
+    billingPage.createBillingProject(billingProjectName, Config.Projects.billingAccount)
+    register cleanUp Rawls.admin.deleteBillingProject(billingProjectName, UserInfo(OAuth2BearerToken(user.makeAuthToken().value), WorkbenchUserId("0"), WorkbenchEmail(user.email), 3600))(UserPool.chooseAdmin.makeAuthToken())
+
+    val statusOption = billingPage.waitForCreateDone(billingProjectName)
+
+    statusOption match {
+      case None | Some("failure") if trials > 1 =>
+        logger.info(s"failure or timeout creating project $billingProjectName, retrying ${trials-1} more times")
```

Comment:
Maybe `warn` rather than `info`.

Reply:
does it really matter?

Reply:
Doubtful. I probably tend to use info when I should use debug or trace so this feels more important to me than those. I don't really care a lot. I should have said "nit".